Gems/AWSClientAuth/Code/Include/Private/Authorization/AWSClientAuthCognitoCachingAuthenticatedCredentialsProvider.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <aws/cognito-identity/CognitoIdentityClient.h>
+#include <aws/identity-management/auth/CognitoCachingCredentialsProvider.h>
+#include <aws/identity-management/auth/PersistentCognitoIdentityProvider.h>
+
+namespace AWSClientAuth
+{
+    // custom implementation to work around https://github.com/aws/aws-sdk-cpp/issues/1448
+    class AWSClientAuthCognitoCachingAuthenticatedCredentialsProvider
+        : public Aws::Auth::CognitoCachingCredentialsProvider
+    {
+    public:
+        AWSClientAuthCognitoCachingAuthenticatedCredentialsProvider(
+            const std::shared_ptr<Aws::Auth::PersistentCognitoIdentityProvider>& identityRepository,
+            const std::shared_ptr<Aws::CognitoIdentity::CognitoIdentityClient>& cognitoIdentityClient = nullptr);
+
+    protected:
+        Aws::CognitoIdentity::Model::GetCredentialsForIdentityOutcome GetCredentialsFromCognito() const override;
+    };
+
+    class AWSClientAuthCachingAnonymousCredsProvider : public AWSClientAuthCognitoCachingAuthenticatedCredentialsProvider
```

Comment:
Curious: Why isn't AWSClientAuthCachingAnonymousCredsProvider derived from Aws::Auth::CognitoCachingCredentialsProvider directly?

Also I think it might be better to define a virtual destructor for AWSClientAuthCognitoCachingAuthenticatedCredentialsProvider to make sure that the derived class can be destructed cleanly: https://stackoverflow.com/questions/461203/when-to-use-virtual-destructors

Reply:
Its not the top level of hierarchy, would assume CognitoCachingCredentialsProvider has virtual destructor. See https://timsong-cpp.github.io/cppwp/n4659/class.dtor#10

It derives eventually from https://github.com/aws/aws-sdk-cpp/blob/bb1fdce01cc7e8ae2fe7162f24c8836e9d3ab0a2/aws-cpp-sdk-core/include/aws/core/auth/AWSCredentialsProvider.h#L50

Does not need an explicit destructor unless we have state or managed resources to clean up.